storage/innobase/sync/srw_lock.cc

```diff
@@ -392,14 +391,22 @@ template void ssux_lock_impl<false>::wr_wait(uint32_t);
 template<bool spinloop>
 void ssux_lock_impl<spinloop>::rd_wait()
 {
-  for (;;)
+  if (spinloop)
   {
-    writer.wr_lock();
-    bool acquired= rd_lock_try();
-    writer.wr_unlock();
-    if (acquired)
-      break;
+    const unsigned delay= srw_pause_delay();
+
+    for (auto spin= srv_n_spin_wait_rounds; spin; spin--)
+    {
+      srw_pause(delay);
+      if (rd_lock_try())
+        return;
+    }
   }
```

Comment:
We might end the `if (spinloop)` body in `this->rd_wait<false>(); return;` to avoid some code duplication.

Reply:
This is not so simple because the template parameter is for the type and no for the member function. Besides, with a change like 79039bdc6f028c86355f0045c70dc3b7c97e05aa no `spinloop=false` variant would need to be instantiated.